### PR TITLE
[WIP] Set emergency login for core user

### PR DESCRIPTION
**Pull request by gbraad**
**Fixes:** #3677

## Solution/Idea
Performs a `passwd` for the `core` user as soon as the VM is reachable, comparable to running the command: 

```
PS> ssh -i c:\users\gbraad\.crc\machines\crc\id_ecdsa core@localhost -p 2222 -C "echo rescue | sudo passwd core --stdin"  
Changing password for user core.
```

The password is fixed to `rescue`.

## Proposed changes

List main as well as consequential changes you introduced or had to introduce.

1. `crc config set enable-emergency-login true`
2. `crc start`
3. open the console and login using: core, rescue

image

### CODE REVIEW

**Comment by gbraad** [comment]
WIP => ~unable to perform cross-builds on Windows. Unable to perform `gofmt` on Windows without modifying almost all the files. Will perform these steps on another machine ASAP.~ For now, testing/debugging the core dumps.

**Comment by cfergeau** [comment]
> The password is fixed to `rescue`.

A non-fixed password is needed in my opinion, especially as this is a persistent option that you can forget to unset after setting it once.

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This configuration file was automatically generated by Gitpod.
```
Comment (cfergeau) [inline .gitpod.yml]: If you want to add this file, please do this in a separate commit.

Comment (gbraad) [reply .gitpod.yml]: oops, wasn't supposed to happen. did the `gofmt` on a remote machine

**Comment by gbraad** [comment]
> A non-fixed password is needed in my opinion

I understand, such as a random generated one. However, this is also a clear choice of the user, so I actually do not think this is such a necessity.

> this is a persistent option

Rather to have this as a flag for `crc start` ? This has crossed my mind

File: .gitpod.yml (hunk 1 above)
Comment (gbraad) [reply .gitpod.yml]: Removed

**Comment by cfergeau** [comment]
> > A non-fixed password is needed in my opinion
> 
> I understand, such as a random generated one. However, this is also a clear choice of the user, so I actually do not think this is such a necessity.
> 

It's a clear choice the user does once, but which then persists forever (I assume most users will forget to unset it). It's not automatically cleared when the VM is deleted, or after a short while. I don't think having a random generated password is going to be a lot more complicated ? (can be 8-10 ascii chars/numbers to make it relatively easy to type)

> > this is a persistent option
> 
> Rather to have this as a flag for `crc start` ? This has crossed my mind

This would ensure it's not persistent, but might be more complicated to add to the codebase.

Another alternative, the config is not a boolean, but the password the user wants to set on the instance? This would ensure the password is not shared/known between all crc deployments. There could even be a magic value (`<<RANDOM>>`) to get a random one to be generated? (this might be going too far ;)

**Comment by gbraad** [comment]
> I don't think having a random generated password is going to be a lot more complicated ? (can be 8-10 ascii chars/numbers to make it relatively easy to type)

More how to handle this. Sure, we can dump it to the stdout... but what if the user does not read this? We store it somewhere? in that case it does not do much better for security

> but the password the user wants to set on the instance

I considered this. It would solve the issue of 'storing' the value, as the user would already be aware of this. Though it still persists and is no safer than storing a file.

**Comment by cfergeau** [comment]
> 

The password does not need to be safer than the ssh key to access the VM, which is stored in a file in ~/.crc/machines/crc. The main thing I want to avoid is all the VMs of all our users sharing the same password (if they enabled the option).

**Comment by gbraad** [comment]
I can change the `bool` to `string` and use it when set, otherwise ignore.

The ssh host is only reachable from localhost anyway, so making it even more secure by obfuscation (randomization) or whatever is not going to help/improve this much more.

For me it is more about messaging. So I would prefer to print this to the command line on start. WDYT?

**Comment by gbraad** [comment]
@crc-org/crc-team do we still want this?

**Comment by praveenkumar** [comment]
> https://github.com/orgs/crc-org/teams/crc-team do we still want this?

@gbraad yes, if you want, I can take over and start making changes as per suggestion

**Comment by gbraad** [comment]
I am OK if you wanna continue on this. The need for this on Windows became unnecessary, however it is still relevant overall.

**Comment by praveenkumar** [comment]
It is superseed by #3755

**Comment by praveenkumar** [comment]
/close

In favor of #3755
